crates/ironrdp-async/src/connector.rs

```diff
@@ -187,10 +187,23 @@ where
     S: FramedWrite + FramedRead,
 {
     buf.clear();
+    let written = single_connect_step_read(framed, connector, buf).await?;
+    single_connect_step_write(framed, buf, written).await
+}
+
+pub async fn single_connect_step_read<S>(
```

Comment:
naming: If I understand correctly, it’s not only about "connecting" anymore, am I right? It’s now a general-purpose helper for anything that is a `Sequence`. I suggest to move that into the `framed` module and rename to a more appropriate name. You can do the same for `single_connect_step_write`.

Reply:
See db346b3c2aaf2be522b55d67a595f88c62cc5b29